model/task_lifecycle.go

```diff
@@ -1141,3 +1159,29 @@ func checkResetDisplayTask(t *task.Task) error {
 
 	return nil
 }
+
+// if part of single-host task group, reset all tasks if finished and should be reset
+func tryResetTaskGroup(t *task.Task) error {
+	if !t.IsPartOfSingleHostTaskGroup() {
+		return nil
+	}
+	tasks, err := GetTasksInTaskGroup(t)
+	if err != nil {
+		return errors.Wrapf(err, "can't get tasks for task group '%s'", t.TaskGroup)
+	}
+
+	resetWhenFinished := false
+	isBlocked := IsBlockedSingleHostTaskGroup(t)
+	for _, taskInGroup := range tasks {
+		if taskInGroup.ResetWhenFinished {
+			resetWhenFinished = true
+		}
+		if !isBlocked && !taskInGroup.IsFinished() {
```

Comment:
`isBlocked` is the result of calling `IsBlockedSingleHostTaskGroup`, which ends up being an analysis of the entire group (and potentially an expensive one). I think therefore it would make sense to move calling `IsBlockedSingleHostTaskGroup` below this for loop (and possibly only after checking `resetWhenFinished`.